design/src/docs/pokemon_service.md

```diff
@@ -0,0 +1,239 @@
+## The Pokémon Service
```

Comment:
Consider stating here upfront that this documentation is meant for contributors to the codebase, and not users.